NitroxClient/GameLogic/Spawning/WorldEntities/PlaceholderGroupWorldEntitySpawner.cs

```diff
@@ -39,55 +73,94 @@ public IEnumerator SpawnAsync(WorldEntity entity, Optional<GameObject> parent, E
+        }
+    }
 
-                    Optional<EntityMetadataProcessor> metadataProcessor = EntityMetadataProcessor.FromMetaData(entity.Metadata);
+    private IEnumerator SpawnChildPlaceholder(PrefabPlaceholder prefabPlaceholder, PrefabPlaceholderEntity entity, TaskResult<Optional<GameObject>> result)
+    {
+        TaskResult<GameObject> goResult = new();
+        yield return DefaultWorldEntitySpawner.CreateGameObject(TechType.None, prefabPlaceholder.prefabClassId, goResult);
 
-                    if (metadataProcessor.HasValue)
-                    {
-                        metadataProcessor.Value.ProcessMetadata(gameObject, placeholder.Metadata);
-                    }
+        if (!goResult.value)
+        {
+            result.Set(Optional.Empty);
+        }
 
-                    break;
-                case WorldEntity slotEntity:
-                    IWorldEntitySpawner spawner = spawnerResolver.ResolveEntitySpawner(slotEntity);
-                    Optional<GameObject> slotEntityParent = Optional.Of(prefabPlaceholder.gameObject);
+        GameObject gameObject = goResult.value;
+        gameObject.transform.SetParent(prefabPlaceholder.transform, false);
+        gameObject.transform.localPosition = Vector3.zero;
+        gameObject.transform.localRotation = Quaternion.identity;
 
-                    TaskResult<Optional<GameObject>> slotEntityTaskResult = new();
-                    yield return spawner.SpawnAsync(slotEntity, slotEntityParent, cellRoot, slotEntityTaskResult);
+        NitroxEntity.SetNewId(gameObject, entity.Id);
 
-                    if (slotEntityTaskResult.value.HasValue)
-                    {
-                        // For some reasons it's not zero after spawning so we reset it here
-                        slotEntityTaskResult.value.Value.transform.localPosition = Vector3.zero;
-                    }
+        EntityMetadataProcessor.ApplyMetadata(gameObject, entity.Metadata);
 
-                    break;
-                default:
-                    Log.Debug(placeholderSlot.ChildEntities.Count > 0 ? $"Unhandled child type {placeholderSlot.ChildEntities[0]}" : "Child was null");
-                    break;
+        result.Set(gameObject);
+    }
+
+    private IEnumerator SpawnWorldEntityChild(WorldEntity worldEntity, EntityCell cellRoot, Optional<GameObject> parent, TaskResult<Optional<GameObject>> worldEntityResult)
+    {
+        IWorldEntitySpawner spawner = spawnerResolver.ResolveEntitySpawner(worldEntity);
+
+        yield return spawner.SpawnAsync(worldEntity, parent, cellRoot, worldEntityResult);
+
+        if (worldEntityResult.value.HasValue)
+        {
+            EntityMetadataProcessor.ApplyMetadata(worldEntityResult.value.Value, worldEntity.Metadata);
+
+            // For some reasons it's not zero after spawning so we reset it here
+            worldEntityResult.value.Value.transform.localPosition = Vector3.zero;
```

Comment:
As clement mentioned in my previous pr he was asking himself if the rotation is also wrong/not zero. I couldn't check this so maybe you could give it a look.

Reply:
I don't personally see anything off with the rotations, should be good.